### PR TITLE
add env for Che in deployment template

### PR DESCRIPTION
**Pull request by xcoulon**
include required env var for the Che service URL in the deployment template.

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @xcoulon for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-services/fabric8-auth/blob/78c7e8e30b87f57b13cf1ed7790890c5a63a311f/.ike-prow/test-keeper.yaml).

File: openshift/auth.app.yaml [hunk 1]
```diff
@@ -169,6 +169,11 @@ objects:
               configMapKeyRef:
                 name: auth
                 key: notification.serviceurl
+          - name: AUTH_CHE_SERVICEURL
+            valueFrom:
+              secretKeyRef:
```
Comment (alexeykazakov) [inline openshift/auth.app.yaml]: Don’t have to be a secret. Use config map instead.

Also we need to add it to https://github.com/fabric8-services/fabric8-auth/blob/master/openshift/auth.config.yaml (just as example for documentation porpoises).

And please add a check for this URL in configuration validation. Similar to https://github.com/fabric8-services/fabric8-auth/blob/master/configuration/configuration.go#L334

The validation will prevent deployments when the ConfigMap is not set.

We also have to create the corresponding CM in preview and prod.

Comment (xcoulon) [reply openshift/auth.app.yaml]: > We also have to create the corresponding CM in preview and prod. 
that's already in progress

> And please add a check for this URL in configuration validation. Similar to https://github.com/fabric8-services/fabric8-auth/blob/master/configuration/configuration.go#L334
hum, so having default values just for testing purpose is not a valid solution, then?

Comment (xcoulon) [reply openshift/auth.app.yaml]: @alexeykazakov I added the `che.serviceurl` CM entry in https://github.com/fabric8-services/fabric8-auth/blob/master/openshift/auth.config.yaml in b0bfeac

**Comment by alexeykazakov** [comment]
[test]

**Pull request merged**